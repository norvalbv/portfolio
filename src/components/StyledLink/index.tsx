import { ReactElement } from 'react';
import { classNames } from '@/src/utils';

type StyledLinkProps = {
  link: string;
  label: string;
  className?: string;
};

const StyledLink = ({ link, label, className }: StyledLinkProps): ReactElement => {
  return (
    <a
      href={link}
      className={classNames(
        'inline-block text-accent-primary underline underline-offset-4 hover:text-accent-secondary',
        className
      )}
      target="_blank"
      rel="noreferrer noopener"
    >
      {label}
    </a>
  );
};

export default StyledLink;
