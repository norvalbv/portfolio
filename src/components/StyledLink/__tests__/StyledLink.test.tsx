import React from 'react';
import renderer from 'react-test-renderer';

import StyledLink from '..';

describe('<StyledLink />', () => {
  test('should render correctly with mandatory props', () => {
    const tree = renderer.create(<StyledLink link="/" label="Test Label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
