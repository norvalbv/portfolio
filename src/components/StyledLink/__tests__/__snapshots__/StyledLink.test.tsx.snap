// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledLink /> should render correctly with mandatory props 1`] = `
<a
  className="text-accent-secondary hover:underline"
  href="/"
  rel="noreferrer noopener"
  target="_blank"
>
  Test Label
</a>
`;
